src/solvers/CMakeLists.txt

```diff
@@ -87,6 +87,15 @@ foreach(SOLVER ${sat_impl})
         target_sources(solvers PRIVATE ${minisat2_source})
 
         target_link_libraries(solvers minisat2-condensed)
+    elseif("${SOLVER}" STREQUAL "system-minisat2")
+        include(CheckIncludeFileCXX)
+        CHECK_INCLUDE_FILE_CXX("minisat/simp/SimpSolver.h" minisat_header_found)
```

Comment:
💡  From CMake 3.12 `CHECK_INCLUDE_FILE_CXX` can also check the existence of a counterpart library component by using `CMAKE_REQUIRED_LIBRARIES`.
It may be worth increasing the minimum cmake required version and check for the library as well.

Reply:
Moving beyond 3.10 would mean CBMC no longer builds (with CMake) on Ubuntu 18.04. Perhaps not a very strong argument as that is long deprecated, but still. So for now I've just added a comment reflecting what you educated me about.

Reply:
> Moving beyond 3.10 would mean CBMC no longer builds (with CMake) on Ubuntu 18.04. Perhaps not a very strong argument as that is long deprecated, but still. So for now I've just added a comment reflecting what you educated me about.

I'd like to keep build on Ubuntu 18.04 alive for a little bit longer -- this is/was an incredibly popular choice, not least for university academics, but well beyond that.

Reply:
It may be worth noting that the Ubuntu 18.04 CI jobs were removed in this PR - https://github.com/diffblue/cbmc/pull/7657

The runners for github actions no longer support Ubuntu 18.04. So we had no straight forward way to keep the checks operational. It could potentially be done, but would require porting them over to docker. As such it is currently only a matter of time before issues are introduced which accidentally break the Ubuntu 18.04 build.